src/common/eeprom.cpp

```diff
@@ -304,15 +304,23 @@ static void eeprom_update_crc32();
 static uint16_t eeprom_fwversion_ui16(void);
 
 // public functions - described in header
+extern void main_preinit();
+
+eeprom_init_status_t eeprom_init(void) {
```

Comment:
please review the initialization procedure again, it doesn't look too safe:
- what happens if you call eeprom_init while initialization in progress (in another thread)?
- what is the meaning of `eeprom_init_status` and its relation to `status`?
- is it possible to call osSemaphoreDef repeatedly, thus leaking the semaphore object in RTOS

Reply:
it it ensured that eeprom_init  is called before FreeRTOS init (before threads even exist)

Reply:
```
int main(void) {
    // initialize eeprom, if it is not already done
    // calls main_preinit()
    eeprom_init_status_t status = eeprom_init();
```